Allow multiple lines on eval

fasfsfgs opened the pull request:
nrepl send method uses promise and returns an array of objects.

Bencode lib changed to allow getting multiple objects from nrepl.

Fixes #9.

Commit:
Allow multiple lines on eval

nrepl send method uses promise and returns an array of objects.

Bencode lib changed to allow getting multiple objects from nrepl.

Fixes #9.

Commit:
handles nrepl response with multiple messages

now we expect multiple messages from a single nrepl data response

changed back to bencoder lib since it nicely makes users aware when the
full buffer didn’t get used

avli commented on src/bencodeUtil.ts:
```diff
@@ -0,0 +1,48 @@
+    try {
+        const decodedObj = bencoder.decode(decodedResult.rest);
+        decodedResult.decodedObjects.push(decodedObj);
+        decodedResult.rest = new Buffer('');
```
@fasfsfgs Accrding to [the documentation](https://nodejs.org/api/buffer.html) it's better to use `Buffer.from('')` for creating an empty buffer.

fasfsfgs replied:
@avli Good advice! I'm going to bump up vscode dev dependency version. The one we use is still on node 4.x which doesn't have those buffer improvements.

avli reviewed (changes requested):
@fasfsfgs Great job! I've left just one comment inside the PR, please, check it out.

And one more thing. Please, ensure you've beautified the code before submitting – it will help us to keep the code style more or less the same. I use [this plugin](https://marketplace.visualstudio.com/items?itemName=HookyQR.beautify) for that.

fasfsfgs commented:
@avli That beautify plugin doesn't support typescript. And when using it in javascript mode it made a lot of changes on the previous code as well.
vscode already does typescript formatting with alt+shift+f but I didn't use that either because it did some changes on the previous code.
Let me know what you think we should do.

Commit:
Update vscode dev lib version

Updated vscode dev libs to the latest recommended settings according to
their development guide.

No need to have typings folder anymore.

avli commented:
@fasfsfgs No problems. Let's use the built in VSCode formatter then.

Commit:
Refactoring: new Buffer to Buffer.from

According to the node documentation it's better to use Buffer.from('')
for creating an empty buffer. (from @avli)
